### PR TITLE
Bug 1205894 - Assign a title to an unnamed attachment

### PR DESCRIPTION
**Pull request by zbraniecki**
https://bugzilla.mozilla.org/show_bug.cgi?id=1205894

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e754e8099a151fb4e8954ace71f53a09c0bba6f1

File: apps/sms/views/conversation/js/attachment.js [hunk 1]
```diff
@@ -73,32 +73,37 @@
...
-
-      var activity = new MozActivity({
-        name: 'open',
-        data: {
-          type: mimetype,
-          filename: filename,
-          blob: this.blob,
-          allowSave: options && options.allowSave
-        }
+      navigator.mozL10n.formatValue('unnamed-attachment').then(unnamed => {
```
Comment (steveck-chung) [inline apps/sms/views/conversation/js/attachment.js]: It will force all the attachment viewing localize the unnamed-attachment string, but not all the attachments will need this step. So how about having a promise like this:
```
var l10nPromise = this.name ? Promise.resolve() :
  navigator.mozL10n.formatValue('unnamed-attachment');

return l10nPromise.then(() => {
  ...
});
```
BTW you can also treat activity as promise since it's DomRequest and catch the error together in l10nPromise reject. But it's optional and you might need to replace the mock activity in attachment unit test.

Comment (zbraniecki) [inline apps/sms/views/conversation/js/attachment.js]: Thanks! I'll update the patch but will not go for transforming the promise chain here to minimize the size of the patch.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d71de62f62535e601adb761a72d48a16b21d0b34

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6f29e9eac2738d42641f2e8a5a7428be1d7b0dad

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e1c43827130817103be94204fa91cc8b50a80133

File: apps/sms/views/conversation/test/unit/attachment_test.js [hunk 2]
```diff
@@ -145,62 +145,67 @@ suite('attachment_test.js', function() {
...
+      attachment.view().then(() => {
+        assert.ok(typeSpy.calledWith(fileName, 'image/jpeg'));
+        assert.ok(matchSpy.calledWith(fileName, typeSpy.returnValues[0]));
+        assert.equal(MockMozActivity.calls.length, 1);
+      }).then(done, done);
     });
 
-    test('Filename has no extension', function() {
+    test('Filename has no extension', function(done) {
       var fileName = 'jpeg_image.jpg';
```
Comment (steveck-chung) [inline apps/sms/views/conversation/test/unit/attachment_test.js]: It's not related to your patch, but this part seems wrong that filename should be `jpeg_image` instead of `jpeg_image.jpg`. Could you please fix this as since you're modifying the attachment unit test? Sorry for the inconvenience... :/

File: apps/sms/views/conversation/test/unit/attachment_test.js [hunk 3]
```diff
@@ -145,62 +145,67 @@ suite('attachment_test.js', function() {
...
+        return attachment2.view();
+      }).then(() => {
+        assert.equal(MockMozActivity.calls.length, 2);
+        assert.equal(
+          MockMozActivity.calls[1].data.filename,
+          'sms-attachments/attachment2.jpg'
+        );
+      }).then(done, done);
 
     });
```
Comment (steveck-chung) [inline apps/sms/views/conversation/test/unit/attachment_test.js]: Could you add one more test about the empty file name case to verify your change for the attachment?

File: apps/sms/views/conversation/js/attachment.js [hunk 4]
```diff
@@ -73,32 +73,37 @@
       //   - Blob mimetype is unsupported but file extension is valid.
```
Comment (steveck-chung) [inline apps/sms/views/conversation/js/attachment.js]: nit: move the comment inside the promise resolve

File: apps/sms/views/conversation/test/unit/attachment_test.js (hunk 2 above)
Comment (zbraniecki) [inline apps/sms/views/conversation/test/unit/attachment_test.js]: sure! :)

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ad4bd60a91a2474c37a48f832535b87dd0782a96

File: apps/sms/views/conversation/js/attachment.js [hunk 5]
```diff
@@ -69,36 +69,42 @@
...
-
-      var activity = new MozActivity({
-        name: 'open',
-        data: {
-          type: mimetype,
-          filename: filename,
-          blob: this.blob,
-          allowSave: options && options.allowSave
-        }
+      // Make sure media is openable and savable even if
```
Comment (steveck-chung) [inline apps/sms/views/conversation/js/attachment.js]: nit: I think you don't have to leave duplicated comment here.
